app/controllers/posts_controller.rb

```diff
@@ -25,4 +25,16 @@ def create
       flash.alert = 'Error!, Post not added!'
     end
   end
+
+  def destroy
+    @post = Post.find(params[:id])
+    authorize! :destroy, @post
+    if @post.destroy
+      flash[:sucess] = 'post was successfully deleted'
+      redirect_to user_posts_path(@post.author)
+    else
+      flash[:error] = 'Failed to delete post'
+      redirect_to user_post_path(@post)
+    end
+  end
```

Comment:
- [ ] as part of the requirements you should use **CanCanCan** for Post authorization in `Post Controller` kindly fix it by adding ` load_and_authorize_resource` in your posts_controller.rb 🙏 

```ruby
class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```